internal-dns/src/resolver.rs

```diff
@@ -69,7 +69,10 @@ impl Resolver {
                 bind_addr: None,
             });
         }
-        let resolver = TokioAsyncResolver::tokio(rc, ResolverOpts::default())?;
+        let mut opts = ResolverOpts::default();
+        opts.use_hosts_file = false;
+        opts.num_concurrent_reqs = dns_addrs.len();
```

Comment:
Weird that this would make a big difference if it already defaults to 2, though I guess if it's always using up its two on the bad ones it's not enough?